src/app/EventHeader.h

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "ConcreteEventPath.h"
+#include "EventLoggingTypes.h"
+#include <app/util/basic-types.h>
+
+namespace chip {
+namespace app {
+struct EventHeader
+{
+    ConcreteEventPath mPath;
+    EventNumber mEventNumber     = 0;
+    PriorityLevel mPriorityLevel = PriorityLevel::Invalid;
+    Timestamp mTimestamp;
+    Timestamp mDeltaTimestamp;
```

Comment:
Deltas are an encoding level detail that is meant for compression over the wire. When it gets up to the application, they should really be seeing a version of this header as defined by the IM spec, namely, `EpochTimeStamp` and `SystemTimeStamp`.

Reply:
Timestamp class has enum class inside, either epoch or system